### PR TITLE
Add Chinese localization

### PR DESCRIPTION
**Pull request by dicksonlaw583**

### CODE REVIEW

**Comment by camillevilla** [comment]
@cbeer has pointed out that we may want to preserve snake case in some keys so that we can take advantage of i18next contexts:
https://www.i18next.com/translation-function/context

So let's just focus on option 1 vs. option 2.

**Comment by dicksonlaw583** [comment]
On a second thought, I noticed on the [documentation for i18next](https://www.i18next.com/overview/configuration-options) that it needs to be something like `zh-CN`, so automated language detection might not work out of the box with either camelCase or snake_case. What is your take on how to name the JSON-containing directories and keys in `settings.js` and `i18n.js` going forward? Use cases similar to this one will certainly show up in other language combinations as well, even English (`en-GB` vs. `en-US`), so it would be best for the team to discuss and decide on a firm policy early.

File: src/config/settings.js [hunk 1]
```diff
@@ -194,6 +194,8 @@ export default {
   availableLanguages: { // All the languages available in the language switcher
     de: 'Deutsch',
     en: 'English',
+    zh_cn: '中文(简体)',
```
[the remark this answers is not in the document]
Comment (dicksonlaw583) [reply src/config/settings.js]: Updated.

File: src/config/settings.js [hunk 2]
```diff
@@ -194,6 +194,8 @@ export default {
...
+    zh_tw: '中文(繁體)',
```
[the remark this answers is not in the document]
Comment (dicksonlaw583) [reply src/config/settings.js]: Updated.

File: src/i18n.js [hunk 3]
```diff
@@ -2,12 +2,16 @@ import i18n from 'i18next';
 import { initReactI18next } from 'react-i18next';
 import de from './locales/de/translation.json';
 import en from './locales/en/translation.json';
+import zh_cn from './locales/zh_cn/translation.json';
```
[the remark this answers is not in the document]
Comment (dicksonlaw583) [reply src/i18n.js]: Updated.

File: src/i18n.js [hunk 4]
```diff
@@ -2,12 +2,16 @@ import i18n from 'i18next';
...
+import zh_tw from './locales/zh_tw/translation.json';
```
[the remark this answers is not in the document]
Comment (dicksonlaw583) [reply src/i18n.js]: Updated.

File: src/i18n.js [hunk 5]
```diff
@@ -2,12 +2,16 @@ import i18n from 'i18next';
...
 
 
 // Load translations for each language
 const resources = {
   de,
   en,
+  zh_cn,
```
[the remark this answers is not in the document]
Comment (dicksonlaw583) [reply src/i18n.js]: Updated.

File: src/i18n.js [hunk 6]
```diff
@@ -2,12 +2,16 @@ import i18n from 'i18next';
...
+  zh_tw,
```
[the remark this answers is not in the document]
Comment (dicksonlaw583) [reply src/i18n.js]: Updated.

**Comment by camillevilla** [comment]
Great! Thank you @dicksonlaw583 👏 I have ticketed up the issue re: snake casing in the `translation.json` keys in #2603

**Pull request merged**